documentation/specs/proposed/RAR_caching_onepager.md

```diff
@@ -0,0 +1,54 @@
+## RAR caching
+
+During every build we need to gather the graph of references and pass
+them to the compiler. This process is highly cacheable as the references
+don’t change all that often. Currently we have some limited caching in
+place however the way nodes are assigned work results in frequent cache
+misses.
+
+### Goals and motivations
+
+1ES team wants to isolate their File I/O related to the which is causing
+issues to their debugging efforts.
+
+Our motivation is a possible performance gain however we’re fine with
+the change as long as the impact is not negative.
+
+### Impact
+
+The only impact we’re concerned about is the performance. There will be
+a tension between the gains from caching and costs due to the IPC from
+the process that will act as the cache repository. We need to ensure
+that this balance will be a net positive performance wise.
+
+### Stakeholders
+
+1ES team, Tomas Bartonek, Rainer Sigwald
+
+1ES team will provide the initial cache implementation. We will review
+their PRs and do the performance evaluations. Handover will be
+successful if nothing breaks and we meet our performance requirements.
```

Comment:
"we meet our performance requirements" - what are they?

Reply:
No regression at the minimum. I mentioned it earlier so I ommited it here. I can remedy.